ctapipe/io/datawriter.py

```diff
@@ -500,6 +515,40 @@ def fill_from_simtel(
                         containers=hist_container,
                     )
 
+    def write_muon_events(self, event: ArrayEventContainer):
```

Comment:
probably should make this private like the other `_write*` functions (rename to `_write_muon_events`).  We may at some point create a public API, but I'd avoid it now.

Reply:
This is called explicitly in stage1

Reply:
I'd second that. However, this would then require to include `_write_muon_events` into datawriter's `__call__`. At the moment, `self.write_dl1.write_muon_events(event)` is called in stage1 to minimize interaction between those parts of the datawriter that write "normal" events and those that write muons, which implies a public function.

Reply:
I think this is a leftover from the different calibration discussion. You cannot have it in `__call__` if you want to be able to apply a different calibration to the muon candidates.

But since we basically removed this discussion here, let's make it consistent and add the muon writing to `__call__`